lib/voyage/templates/admin_application_controller.rb

```diff
@@ -0,0 +1,31 @@
+# All Administrate controllers inherit from this `Admin::ApplicationController`,
+# making it the ideal place to put authentication logic or other
+# before_actions.
+#
+# If you want to add pagination or other controller-level concerns,
+# you're free to overwrite the RESTful controller actions.
+module Admin
+  class ApplicationController < Administrate::ApplicationController
+    include AnalyticsTrack
+
+    before_action :authenticate_user!,
+                  unless: -> { is_a?(HighVoltage::PagesController) }
+    before_action :require_admin!
+
+    skip_authorization_check
+    impersonates :user
```

Comment:
Given that we're baking in impersonates, could we include the views in the admin to do the impersonation?

Reply:
Good point. In backporting these controller changes, I missed the respective Administrate view customizations as well.